### PR TITLE
CloudPrivateIPConfig reconcile: don't throw err on missing annotation

### PR DESCRIPTION
**Pull request by andreaskaris**
Do not throw an error if the owner annotation if missing. Instead log it and return nil. Otherwise there's a big risk that the controller loops on the error for a very long time.

It makes no sense to throw an error upon deletion if the annotation is missing: the deleted object will not get the annotation any more. It makes no sense to throw an error upon add if the annotation is missing: if this is a new object then the there will be an update action when the object is updated so we should retrieve the annotation on update.

Reported-at: https://issues.redhat.com/browse/OCPBUGS-4500
Signed-off-by: Andreas Karis <ak.karis@gmail.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/egressip.go [hunk 1]
```diff
@@ -599,7 +599,12 @@ func (oc *DefaultNetworkController) reconcileCloudPrivateIPConfig(old, new *ocpc
 		// Get the EgressIP owner reference
 		egressIPName, exists := oldCloudPrivateIPConfig.Annotations[util.OVNEgressIPOwnerRefLabel]
 		if !exists {
-			return fmt.Errorf("CloudPrivateIPConfig object: %s is missing the egress IP owner reference annotation", oldCloudPrivateIPConfig.Name)
+			// If a CloudPrivateIPConfig object does not have an egress IP owner reference annotation upon deletion,
+			// there is no way that the object will get one after deletion. Hence, simply log a warning message here
+			// for informative purposes instead of throwing the same error and retrying time and time again.
+			klog.Warningf("CloudPrivateIPConfig object %q was missing the egress IP owner reference annotation "+
```
Comment (kyrtapz) [inline go-controller/pkg/ovn/egressip.go:605]: I think this should be an error log.

**Comment by kyrtapz** [comment]
Not returning an error in those two places only disables the retry mechanism.
As mentioned in the comment there is no point of retrying when the annotation is missing.

/lgtm

File: go-controller/pkg/ovn/egressip.go (hunk 1 above)
Comment (andreaskaris) [reply go-controller/pkg/ovn/egressip.go:605]: I'll keep it at the Warning message if you don't mind; if someone manually removes the annotation from the CloudPrivateIPConfig, then I think we can suppose that it's intentional. Indeed, in those situations, the EgressIP will be deleted but without the owner annotation on the CloudPrivateIPConfig, the latter will not be removed

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Review by jcaamano — APPROVED** [review]

**Pull request merged**